### ISSUE TITLE
Remove setters from deletion and modification audit interfaces

### ISSUE THREAD

**Issue by Trojaner** [abpframework/abp#12229]
For example; IHasCreationTime has no public setter but ISoftDelete and IHasDeletionTime have public setters.

I suggest to remove all setters and use ObjectHelper.TrySetProperty internally.

**Comment by gterdem** [comment]
Inconsistency between `IHasCreationTime` and `IHasDeletionTime` makes sense since they are audit properties and should be set without user interaction.

However, I believe; filters like `ISoftDelete` should have public setter. We shouldn't limit users about user interactive properties and let the set the object state as they please.

**Comment by hikalkan** [comment]
Actually, only `CreationTime` and `CreatorId` has no public setter. Because, these shouldn't be changed after entity creation (I suppose this is a very common case, you never need to manually set them later). However, ABP doesn't restrict you to have a public setter for these.

For modification and deletion, we haven't thought such a restriction. But you are right, audit interfaces are unnecessarily forces entity classes to have public setters for these properties. We should remove public setters and use `ObjectHelper.TrySetProperty` instead. If anyone wants public setters, they can still make it. But, in this way, we don't force to have public setters.

Basically, we will remove setters from `IModificationAuditedObject`, `IModificationAuditedObject<TUser>`, `IHasModificationTime`, `IHasDeletionTime`, `ISoftDelete` (add more if I forgot it).

However, `AuditedEntity` and other base entity classes will continue to have public setters. If you want to make them private, don't derive from these base classes, but implement the interfaces yourself.
